internal/archiver/archiver.go

```diff
@@ -445,40 +453,24 @@ func (arch *Archiver) Save(ctx context.Context, snPath, target string, previous
 	return fn, false, nil
 }
 
-// fileChanged returns true if the file's content has changed since the node
-// was created.
-func fileChanged(fi os.FileInfo, node *restic.Node, ignoreInode bool) bool {
-	if node == nil {
+// fileChanged tries to detect whether a file's content has changed compared
+// to the contents of node, which describes the same path in the parent backup.
+// It should only be run for regular files.
+func fileChanged(fi os.FileInfo, node *restic.Node, ignore ChangeIgnore) bool {
+	// Easy cases: type has to be "file", size and mtime must always match.
+	if node == nil ||
+		node.Type != "file" ||
+		uint64(fi.Size()) != node.Size ||
+		!fi.ModTime().Equal(node.ModTime) {
```

Comment:
Personally, I would prefer a `switch` here, but this is only, well, personal preference :wink:

Reply:
You're right, that's way more readable. Fixed.